// @aws-sdk/client-dynamodb-v2-node/model/_ReplicaSettingsDescription.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var _AutoScalingSettingsDescription_1 = require("./_AutoScalingSettingsDescription");
var _ReplicaGlobalSecondaryIndexSettingsDescriptionList_1 = require("./_ReplicaGlobalSecondaryIndexSettingsDescriptionList");
exports._ReplicaSettingsDescription = {
    type: 'structure',
    required: [
        'RegionName',
    ],
    members: {
        RegionName: {
            shape: {
                type: 'string',
            },
        },
        ReplicaStatus: {
            shape: {
                type: 'string',
            },
        },
        ReplicaProvisionedReadCapacityUnits: {
            shape: {
                type: 'integer',
                min: 1,
            },
        },
        ReplicaProvisionedReadCapacityAutoScalingSettings: {
            shape: _AutoScalingSettingsDescription_1._AutoScalingSettingsDescription,
        },
        ReplicaProvisionedWriteCapacityUnits: {
            shape: {
                type: 'integer',
                min: 1,
            },
        },
        ReplicaProvisionedWriteCapacityAutoScalingSettings: {
            shape: _AutoScalingSettingsDescription_1._AutoScalingSettingsDescription,
        },
        ReplicaGlobalSecondaryIndexSettings: {
            shape: _ReplicaGlobalSecondaryIndexSettingsDescriptionList_1._ReplicaGlobalSecondaryIndexSettingsDescriptionList,
        },
    },
};
