// @aws-sdk/client-dynamodb-v2-node/model/_SourceTableDetails.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var _KeySchema_1 = require("./_KeySchema");
var _ProvisionedThroughput_1 = require("./_ProvisionedThroughput");
exports._SourceTableDetails = {
    type: 'structure',
    required: [
        'TableName',
        'TableId',
        'KeySchema',
        'TableCreationDateTime',
        'ProvisionedThroughput',
    ],
    members: {
        TableName: {
            shape: {
                type: 'string',
                min: 3,
            },
        },
        TableId: {
            shape: {
                type: 'string',
            },
        },
        TableArn: {
            shape: {
                type: 'string',
            },
        },
        TableSizeBytes: {
            shape: {
                type: 'integer',
            },
        },
        KeySchema: {
            shape: _KeySchema_1._KeySchema,
        },
        TableCreationDateTime: {
            shape: {
                type: 'timestamp',
            },
        },
        ProvisionedThroughput: {
            shape: _ProvisionedThroughput_1._ProvisionedThroughput,
        },
        ItemCount: {
            shape: {
                type: 'integer',
            },
        },
    },
};
