// @aws-sdk/client-dynamodb-v2-node/model/_SourceTableFeatureDetails.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var _LocalSecondaryIndexes_1 = require("./_LocalSecondaryIndexes");
var _GlobalSecondaryIndexes_1 = require("./_GlobalSecondaryIndexes");
var _StreamSpecification_1 = require("./_StreamSpecification");
var _TimeToLiveDescription_1 = require("./_TimeToLiveDescription");
var _SSEDescription_1 = require("./_SSEDescription");
exports._SourceTableFeatureDetails = {
    type: 'structure',
    required: [],
    members: {
        LocalSecondaryIndexes: {
            shape: _LocalSecondaryIndexes_1._LocalSecondaryIndexes,
        },
        GlobalSecondaryIndexes: {
            shape: _GlobalSecondaryIndexes_1._GlobalSecondaryIndexes,
        },
        StreamDescription: {
            shape: _StreamSpecification_1._StreamSpecification,
        },
        TimeToLiveDescription: {
            shape: _TimeToLiveDescription_1._TimeToLiveDescription,
        },
        SSEDescription: {
            shape: _SSEDescription_1._SSEDescription,
        },
    },
};
