// @aws-sdk/client-dynamodb-v2-node/model/CreateTable.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var CreateTableInput_1 = require("./CreateTableInput");
var CreateTableOutput_1 = require("./CreateTableOutput");
var ResourceInUseException_1 = require("./ResourceInUseException");
var LimitExceededException_1 = require("./LimitExceededException");
var InternalServerError_1 = require("./InternalServerError");
var ServiceMetadata_1 = require("./ServiceMetadata");
exports.CreateTable = {
    metadata: ServiceMetadata_1.ServiceMetadata,
    name: 'CreateTable',
    http: {
        method: 'POST',
        requestUri: '/',
    },
    input: {
        shape: CreateTableInput_1.CreateTableInput,
    },
    output: {
        shape: CreateTableOutput_1.CreateTableOutput,
    },
    errors: [
        {
            shape: ResourceInUseException_1.ResourceInUseException,
        },
        {
            shape: LimitExceededException_1.LimitExceededException,
        },
        {
            shape: InternalServerError_1.InternalServerError,
        },
    ],
};
