// @aws-sdk/client-dynamodb-v2-node/model/GetItem.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var GetItemInput_1 = require("./GetItemInput");
var GetItemOutput_1 = require("./GetItemOutput");
var ProvisionedThroughputExceededException_1 = require("./ProvisionedThroughputExceededException");
var ResourceNotFoundException_1 = require("./ResourceNotFoundException");
var InternalServerError_1 = require("./InternalServerError");
var ServiceMetadata_1 = require("./ServiceMetadata");
exports.GetItem = {
    metadata: ServiceMetadata_1.ServiceMetadata,
    name: 'GetItem',
    http: {
        method: 'POST',
        requestUri: '/',
    },
    input: {
        shape: GetItemInput_1.GetItemInput,
    },
    output: {
        shape: GetItemOutput_1.GetItemOutput,
    },
    errors: [
        {
            shape: ProvisionedThroughputExceededException_1.ProvisionedThroughputExceededException,
        },
        {
            shape: ResourceNotFoundException_1.ResourceNotFoundException,
        },
        {
            shape: InternalServerError_1.InternalServerError,
        },
    ],
};
