// @aws-sdk/client-dynamodb-v2-node/model/PutItemInput.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var _PutItemInputAttributeMap_1 = require("./_PutItemInputAttributeMap");
var _ExpectedAttributeMap_1 = require("./_ExpectedAttributeMap");
var _ExpressionAttributeNameMap_1 = require("./_ExpressionAttributeNameMap");
var _ExpressionAttributeValueMap_1 = require("./_ExpressionAttributeValueMap");
exports.PutItemInput = {
    type: 'structure',
    required: [
        'TableName',
        'Item',
    ],
    members: {
        TableName: {
            shape: {
                type: 'string',
                min: 3,
            },
        },
        Item: {
            shape: _PutItemInputAttributeMap_1._PutItemInputAttributeMap,
        },
        Expected: {
            shape: _ExpectedAttributeMap_1._ExpectedAttributeMap,
        },
        ReturnValues: {
            shape: {
                type: 'string',
            },
        },
        ReturnConsumedCapacity: {
            shape: {
                type: 'string',
            },
        },
        ReturnItemCollectionMetrics: {
            shape: {
                type: 'string',
            },
        },
        ConditionalOperator: {
            shape: {
                type: 'string',
            },
        },
        ConditionExpression: {
            shape: {
                type: 'string',
            },
        },
        ExpressionAttributeNames: {
            shape: _ExpressionAttributeNameMap_1._ExpressionAttributeNameMap,
        },
        ExpressionAttributeValues: {
            shape: _ExpressionAttributeValueMap_1._ExpressionAttributeValueMap,
        },
    },
};
