// @aws-sdk/client-dynamodb-v2-node/model/UpdateTableInput.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var _AttributeDefinitions_1 = require("./_AttributeDefinitions");
var _ProvisionedThroughput_1 = require("./_ProvisionedThroughput");
var _GlobalSecondaryIndexUpdateList_1 = require("./_GlobalSecondaryIndexUpdateList");
var _StreamSpecification_1 = require("./_StreamSpecification");
var _SSESpecification_1 = require("./_SSESpecification");
exports.UpdateTableInput = {
    type: 'structure',
    required: [
        'TableName',
    ],
    members: {
        AttributeDefinitions: {
            shape: _AttributeDefinitions_1._AttributeDefinitions,
        },
        TableName: {
            shape: {
                type: 'string',
                min: 3,
            },
        },
        ProvisionedThroughput: {
            shape: _ProvisionedThroughput_1._ProvisionedThroughput,
        },
        GlobalSecondaryIndexUpdates: {
            shape: _GlobalSecondaryIndexUpdateList_1._GlobalSecondaryIndexUpdateList,
        },
        StreamSpecification: {
            shape: _StreamSpecification_1._StreamSpecification,
        },
        SSESpecification: {
            shape: _SSESpecification_1._SSESpecification,
        },
    },
};
