// @aws-sdk/client-dynamodb-v2-node/model/UpdateItem.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var UpdateItemInput_1 = require("./UpdateItemInput");
var UpdateItemOutput_1 = require("./UpdateItemOutput");
var ConditionalCheckFailedException_1 = require("./ConditionalCheckFailedException");
var ProvisionedThroughputExceededException_1 = require("./ProvisionedThroughputExceededException");
var ResourceNotFoundException_1 = require("./ResourceNotFoundException");
var ItemCollectionSizeLimitExceededException_1 = require("./ItemCollectionSizeLimitExceededException");
var InternalServerError_1 = require("./InternalServerError");
var ServiceMetadata_1 = require("./ServiceMetadata");
exports.UpdateItem = {
    metadata: ServiceMetadata_1.ServiceMetadata,
    name: 'UpdateItem',
    http: {
        method: 'POST',
        requestUri: '/',
    },
    input: {
        shape: UpdateItemInput_1.UpdateItemInput,
    },
    output: {
        shape: UpdateItemOutput_1.UpdateItemOutput,
    },
    errors: [
        {
            shape: ConditionalCheckFailedException_1.ConditionalCheckFailedException,
        },
        {
            shape: ProvisionedThroughputExceededException_1.ProvisionedThroughputExceededException,
        },
        {
            shape: ResourceNotFoundException_1.ResourceNotFoundException,
        },
        {
            shape: ItemCollectionSizeLimitExceededException_1.ItemCollectionSizeLimitExceededException,
        },
        {
            shape: InternalServerError_1.InternalServerError,
        },
    ],
};
