// @aws-sdk/client-dynamodb-v2-node/model/_AttributeValue.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var _StringSetAttributeValue_1 = require("./_StringSetAttributeValue");
var _NumberSetAttributeValue_1 = require("./_NumberSetAttributeValue");
var _BinarySetAttributeValue_1 = require("./_BinarySetAttributeValue");
var _MapAttributeValue_1 = require("./_MapAttributeValue");
var _ListAttributeValue_1 = require("./_ListAttributeValue");
exports._AttributeValue = {
    type: 'structure',
    required: [],
    members: {
        S: {
            shape: {
                type: 'string',
            },
        },
        N: {
            shape: {
                type: 'string',
            },
        },
        B: {
            shape: {
                type: 'blob',
            },
        },
        SS: {
            shape: _StringSetAttributeValue_1._StringSetAttributeValue,
        },
        NS: {
            shape: _NumberSetAttributeValue_1._NumberSetAttributeValue,
        },
        BS: {
            shape: _BinarySetAttributeValue_1._BinarySetAttributeValue,
        },
        M: {
            shape: _MapAttributeValue_1._MapAttributeValue,
        },
        L: {
            shape: _ListAttributeValue_1._ListAttributeValue,
        },
        NULL: {
            shape: {
                type: 'boolean',
            },
        },
        BOOL: {
            shape: {
                type: 'boolean',
            },
        },
    },
};
