// @aws-sdk/client-dynamodb-v2-node/model/_BackupDetails.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports._BackupDetails = {
    type: 'structure',
    required: [
        'BackupArn',
        'BackupName',
        'BackupStatus',
        'BackupType',
        'BackupCreationDateTime',
    ],
    members: {
        BackupArn: {
            shape: {
                type: 'string',
                min: 37,
            },
        },
        BackupName: {
            shape: {
                type: 'string',
                min: 3,
            },
        },
        BackupSizeBytes: {
            shape: {
                type: 'integer',
            },
        },
        BackupStatus: {
            shape: {
                type: 'string',
            },
        },
        BackupType: {
            shape: {
                type: 'string',
            },
        },
        BackupCreationDateTime: {
            shape: {
                type: 'timestamp',
            },
        },
        BackupExpiryDateTime: {
            shape: {
                type: 'timestamp',
            },
        },
    },
};
