// @aws-sdk/client-dynamodb-v2-node/model/_ReplicaGlobalSecondaryIndexSettingsDescription.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var _AutoScalingSettingsDescription_1 = require("./_AutoScalingSettingsDescription");
exports._ReplicaGlobalSecondaryIndexSettingsDescription = {
    type: 'structure',
    required: [
        'IndexName',
    ],
    members: {
        IndexName: {
            shape: {
                type: 'string',
                min: 3,
            },
        },
        IndexStatus: {
            shape: {
                type: 'string',
            },
        },
        ProvisionedReadCapacityUnits: {
            shape: {
                type: 'integer',
                min: 1,
            },
        },
        ProvisionedReadCapacityAutoScalingSettings: {
            shape: _AutoScalingSettingsDescription_1._AutoScalingSettingsDescription,
        },
        ProvisionedWriteCapacityUnits: {
            shape: {
                type: 'integer',
                min: 1,
            },
        },
        ProvisionedWriteCapacityAutoScalingSettings: {
            shape: _AutoScalingSettingsDescription_1._AutoScalingSettingsDescription,
        },
    },
};
