// @aws-sdk/client-dynamodb-v2-node/model/CreateGlobalTable.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var CreateGlobalTableInput_1 = require("./CreateGlobalTableInput");
var CreateGlobalTableOutput_1 = require("./CreateGlobalTableOutput");
var LimitExceededException_1 = require("./LimitExceededException");
var InternalServerError_1 = require("./InternalServerError");
var GlobalTableAlreadyExistsException_1 = require("./GlobalTableAlreadyExistsException");
var TableNotFoundException_1 = require("./TableNotFoundException");
var ServiceMetadata_1 = require("./ServiceMetadata");
exports.CreateGlobalTable = {
    metadata: ServiceMetadata_1.ServiceMetadata,
    name: 'CreateGlobalTable',
    http: {
        method: 'POST',
        requestUri: '/',
    },
    input: {
        shape: CreateGlobalTableInput_1.CreateGlobalTableInput,
    },
    output: {
        shape: CreateGlobalTableOutput_1.CreateGlobalTableOutput,
    },
    errors: [
        {
            shape: LimitExceededException_1.LimitExceededException,
        },
        {
            shape: InternalServerError_1.InternalServerError,
        },
        {
            shape: GlobalTableAlreadyExistsException_1.GlobalTableAlreadyExistsException,
        },
        {
            shape: TableNotFoundException_1.TableNotFoundException,
        },
    ],
};
