// @aws-sdk/client-dynamodb-v2-node/model/DeleteBackup.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var DeleteBackupInput_1 = require("./DeleteBackupInput");
var DeleteBackupOutput_1 = require("./DeleteBackupOutput");
var BackupNotFoundException_1 = require("./BackupNotFoundException");
var BackupInUseException_1 = require("./BackupInUseException");
var LimitExceededException_1 = require("./LimitExceededException");
var InternalServerError_1 = require("./InternalServerError");
var ServiceMetadata_1 = require("./ServiceMetadata");
exports.DeleteBackup = {
    metadata: ServiceMetadata_1.ServiceMetadata,
    name: 'DeleteBackup',
    http: {
        method: 'POST',
        requestUri: '/',
    },
    input: {
        shape: DeleteBackupInput_1.DeleteBackupInput,
    },
    output: {
        shape: DeleteBackupOutput_1.DeleteBackupOutput,
    },
    errors: [
        {
            shape: BackupNotFoundException_1.BackupNotFoundException,
        },
        {
            shape: BackupInUseException_1.BackupInUseException,
        },
        {
            shape: LimitExceededException_1.LimitExceededException,
        },
        {
            shape: InternalServerError_1.InternalServerError,
        },
    ],
};
