// @aws-sdk/client-dynamodb-v2-node/model/DeleteItemInput.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var _Key_1 = require("./_Key");
var _ExpectedAttributeMap_1 = require("./_ExpectedAttributeMap");
var _ExpressionAttributeNameMap_1 = require("./_ExpressionAttributeNameMap");
var _ExpressionAttributeValueMap_1 = require("./_ExpressionAttributeValueMap");
exports.DeleteItemInput = {
    type: 'structure',
    required: [
        'TableName',
        'Key',
    ],
    members: {
        TableName: {
            shape: {
                type: 'string',
                min: 3,
            },
        },
        Key: {
            shape: _Key_1._Key,
        },
        Expected: {
            shape: _ExpectedAttributeMap_1._ExpectedAttributeMap,
        },
        ConditionalOperator: {
            shape: {
                type: 'string',
            },
        },
        ReturnValues: {
            shape: {
                type: 'string',
            },
        },
        ReturnConsumedCapacity: {
            shape: {
                type: 'string',
            },
        },
        ReturnItemCollectionMetrics: {
            shape: {
                type: 'string',
            },
        },
        ConditionExpression: {
            shape: {
                type: 'string',
            },
        },
        ExpressionAttributeNames: {
            shape: _ExpressionAttributeNameMap_1._ExpressionAttributeNameMap,
        },
        ExpressionAttributeValues: {
            shape: _ExpressionAttributeValueMap_1._ExpressionAttributeValueMap,
        },
    },
};
