// @aws-sdk/client-dynamodb-v2-node/model/RestoreTableToPointInTime.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var RestoreTableToPointInTimeInput_1 = require("./RestoreTableToPointInTimeInput");
var RestoreTableToPointInTimeOutput_1 = require("./RestoreTableToPointInTimeOutput");
var TableAlreadyExistsException_1 = require("./TableAlreadyExistsException");
var TableNotFoundException_1 = require("./TableNotFoundException");
var TableInUseException_1 = require("./TableInUseException");
var LimitExceededException_1 = require("./LimitExceededException");
var InvalidRestoreTimeException_1 = require("./InvalidRestoreTimeException");
var PointInTimeRecoveryUnavailableException_1 = require("./PointInTimeRecoveryUnavailableException");
var InternalServerError_1 = require("./InternalServerError");
var ServiceMetadata_1 = require("./ServiceMetadata");
exports.RestoreTableToPointInTime = {
    metadata: ServiceMetadata_1.ServiceMetadata,
    name: 'RestoreTableToPointInTime',
    http: {
        method: 'POST',
        requestUri: '/',
    },
    input: {
        shape: RestoreTableToPointInTimeInput_1.RestoreTableToPointInTimeInput,
    },
    output: {
        shape: RestoreTableToPointInTimeOutput_1.RestoreTableToPointInTimeOutput,
    },
    errors: [
        {
            shape: TableAlreadyExistsException_1.TableAlreadyExistsException,
        },
        {
            shape: TableNotFoundException_1.TableNotFoundException,
        },
        {
            shape: TableInUseException_1.TableInUseException,
        },
        {
            shape: LimitExceededException_1.LimitExceededException,
        },
        {
            shape: InvalidRestoreTimeException_1.InvalidRestoreTimeException,
        },
        {
            shape: PointInTimeRecoveryUnavailableException_1.PointInTimeRecoveryUnavailableException,
        },
        {
            shape: InternalServerError_1.InternalServerError,
        },
    ],
};
