// @aws-sdk/client-dynamodb-v2-node/model/TagResource.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var TagResourceInput_1 = require("./TagResourceInput");
var TagResourceOutput_1 = require("./TagResourceOutput");
var LimitExceededException_1 = require("./LimitExceededException");
var ResourceNotFoundException_1 = require("./ResourceNotFoundException");
var InternalServerError_1 = require("./InternalServerError");
var ResourceInUseException_1 = require("./ResourceInUseException");
var ServiceMetadata_1 = require("./ServiceMetadata");
exports.TagResource = {
    metadata: ServiceMetadata_1.ServiceMetadata,
    name: 'TagResource',
    http: {
        method: 'POST',
        requestUri: '/',
    },
    input: {
        shape: TagResourceInput_1.TagResourceInput,
    },
    output: {
        shape: TagResourceOutput_1.TagResourceOutput,
    },
    errors: [
        {
            shape: LimitExceededException_1.LimitExceededException,
        },
        {
            shape: ResourceNotFoundException_1.ResourceNotFoundException,
        },
        {
            shape: InternalServerError_1.InternalServerError,
        },
        {
            shape: ResourceInUseException_1.ResourceInUseException,
        },
    ],
};
