// @aws-sdk/client-dynamodb-v2-node/model/UpdateGlobalTableSettingsInput.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var _AutoScalingSettingsUpdate_1 = require("./_AutoScalingSettingsUpdate");
var _GlobalTableGlobalSecondaryIndexSettingsUpdateList_1 = require("./_GlobalTableGlobalSecondaryIndexSettingsUpdateList");
var _ReplicaSettingsUpdateList_1 = require("./_ReplicaSettingsUpdateList");
exports.UpdateGlobalTableSettingsInput = {
    type: 'structure',
    required: [
        'GlobalTableName',
    ],
    members: {
        GlobalTableName: {
            shape: {
                type: 'string',
                min: 3,
            },
        },
        GlobalTableProvisionedWriteCapacityUnits: {
            shape: {
                type: 'integer',
                min: 1,
            },
        },
        GlobalTableProvisionedWriteCapacityAutoScalingSettingsUpdate: {
            shape: _AutoScalingSettingsUpdate_1._AutoScalingSettingsUpdate,
        },
        GlobalTableGlobalSecondaryIndexSettingsUpdate: {
            shape: _GlobalTableGlobalSecondaryIndexSettingsUpdateList_1._GlobalTableGlobalSecondaryIndexSettingsUpdateList,
        },
        ReplicaSettingsUpdate: {
            shape: _ReplicaSettingsUpdateList_1._ReplicaSettingsUpdateList,
        },
    },
};
