// @aws-sdk/client-dynamodb-v2-node/model/UpdateGlobalTableSettings.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var UpdateGlobalTableSettingsInput_1 = require("./UpdateGlobalTableSettingsInput");
var UpdateGlobalTableSettingsOutput_1 = require("./UpdateGlobalTableSettingsOutput");
var GlobalTableNotFoundException_1 = require("./GlobalTableNotFoundException");
var ReplicaNotFoundException_1 = require("./ReplicaNotFoundException");
var IndexNotFoundException_1 = require("./IndexNotFoundException");
var LimitExceededException_1 = require("./LimitExceededException");
var ResourceInUseException_1 = require("./ResourceInUseException");
var InternalServerError_1 = require("./InternalServerError");
var ServiceMetadata_1 = require("./ServiceMetadata");
exports.UpdateGlobalTableSettings = {
    metadata: ServiceMetadata_1.ServiceMetadata,
    name: 'UpdateGlobalTableSettings',
    http: {
        method: 'POST',
        requestUri: '/',
    },
    input: {
        shape: UpdateGlobalTableSettingsInput_1.UpdateGlobalTableSettingsInput,
    },
    output: {
        shape: UpdateGlobalTableSettingsOutput_1.UpdateGlobalTableSettingsOutput,
    },
    errors: [
        {
            shape: GlobalTableNotFoundException_1.GlobalTableNotFoundException,
        },
        {
            shape: ReplicaNotFoundException_1.ReplicaNotFoundException,
        },
        {
            shape: IndexNotFoundException_1.IndexNotFoundException,
        },
        {
            shape: LimitExceededException_1.LimitExceededException,
        },
        {
            shape: ResourceInUseException_1.ResourceInUseException,
        },
        {
            shape: InternalServerError_1.InternalServerError,
        },
    ],
};
