// @aws-sdk/client-dynamodb-v2-node/model/UpdateTimeToLive.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var UpdateTimeToLiveInput_1 = require("./UpdateTimeToLiveInput");
var UpdateTimeToLiveOutput_1 = require("./UpdateTimeToLiveOutput");
var ResourceInUseException_1 = require("./ResourceInUseException");
var ResourceNotFoundException_1 = require("./ResourceNotFoundException");
var LimitExceededException_1 = require("./LimitExceededException");
var InternalServerError_1 = require("./InternalServerError");
var ServiceMetadata_1 = require("./ServiceMetadata");
exports.UpdateTimeToLive = {
    metadata: ServiceMetadata_1.ServiceMetadata,
    name: 'UpdateTimeToLive',
    http: {
        method: 'POST',
        requestUri: '/',
    },
    input: {
        shape: UpdateTimeToLiveInput_1.UpdateTimeToLiveInput,
    },
    output: {
        shape: UpdateTimeToLiveOutput_1.UpdateTimeToLiveOutput,
    },
    errors: [
        {
            shape: ResourceInUseException_1.ResourceInUseException,
        },
        {
            shape: ResourceNotFoundException_1.ResourceNotFoundException,
        },
        {
            shape: LimitExceededException_1.LimitExceededException,
        },
        {
            shape: InternalServerError_1.InternalServerError,
        },
    ],
};
